addons/mail/static/src/discuss/core/web/thread_actions.js

```diff
@@ -15,14 +15,11 @@ threadActionsRegistry.add("expand-discuss", {
     name: _t("Open in Discuss"),
     open(component) {
         component.thread.setAsDiscussThread();
-        component.actionService.doAction(
-            {
-                type: "ir.actions.client",
-                tag: "mail.action_discuss",
-                name: _t("Discuss"),
-            },
-            { clearBreadcrumbs: true }
-        );
+        component.actionService.doAction({
+            type: "ir.actions.client",
+            tag: "mail.action_discuss",
+            name: component.thread.displayName || _t("Discuss"),
```

Comment:
Did you notice an issue with thread having to name to need a default?

Reply:
I asked Alex if there was any guarantee that a thread has a name and he told me no. I assumed we'd rather have "Discuss" displayed than "undefined" or "null" in that specific case.

Reply:
Then the condition should be added also on the other usage of `displayName`.

But what about leaving it undefined? Won't it use the default name defined on the action instead?

Reply:
Note: assuming that is true, using `onRendered` might not be enough, as when receiving the `displayName` from the server (or info allowing to compute it), there might be no render and we'd still want to update the title.

Reply:
Ok so I checked and 1) it was not needed to add `|| _t("Discuss")` as it uses the action name if displayName is undefined and 2) there is a `useState` in the component around store and I believe it is enough to trigger a re-render whenever this.store.discuss.thread is modified.